doc/installing.rst

```diff
@@ -4,182 +4,181 @@ Installation
+
+        conda install -c conda-forge pyqt
+
+or
 
-        conda env create -f environment.yml
-        conda activate pypeit
-        # install PySide2/PyQt5 from conda/pip using using ONE of the following lines
         conda install -c conda-forge pyside2
-        conda install pyqt
-        pip install pyside2
-        pip install pyqt5
 
- #. Install PypeIt::
+Developer Install via ``pip``
+-----------------------------
+
+It is also possible to install pre-release or development versions of ``PypeIt`` directly from `GitHub <https://github.com/pypeit/PypeIt>`_
+using ``pip``. If you already have a ``pypeit`` environment setup, do:
+
+        pip install --upgrade git+https://github.com/pypeit/PypeIt#egg=pypeit
+
+If you're installing in a clean environment, be sure to include the optional dependencies as well:
+
+        pip install --upgrade git+https://github.com/pypeit/PypeIt#egg=pypeit[pyqt5,pyside2,shapely]
+
+Those commands will install the default branch, ``master``. You can also specify the branch you wish to use:
+
+        pip install --upgrade git+https://github.com/pypeit/PypeIt.git@develop#egg=pypeit[pyqt5,pyside2,shapely]
+
+Commit hashes, tag names, or git refs can also be specified. See the `VCS Support documentation
+<https://pip.pypa.io/en/stable/reference/pip_install/#vcs-support>`_ for details and examples.
+
+Developers doing code development will likely want to set up an "editable" install that points to a locally checked out
+copy of the GitHub repository. It is highly recommened to use ``pip`` for this as well so that the dependencies can be managed.
+It is also recommended to install all optional dependencies within the environment used for ``PypeIt`` development. First,
+clone the repository:
+
+        git clone https://github.com/pypeit/PypeIt.git
 
-        pip install -e .
+Then perform the install, preferably in a clean environment:
 
-Installing the code this way ensures that virtually all changes to files in
-the ``PypeIt`` directory take immediate effect the next time you
-import the code.
+        cd PypeIt
+        pip install -e .[pyqt5,pyside2,shapely,test,docs]
+
+An "editable" install means that any changes you make in that code tree will become immediately available the next
+time the code is imported.
+
+If any of this fails, please `submit an issue
+<https://github.com/pypeit/PypeIt/issues>`__.
+
+
+Dependency Caveats
+------------------
+
+Some users have run into the following complications when installing the
+``PypeIt`` dependencies:
+
+ - Installation of ``numba`` has been known to fault because of an issue with
+   the LLVM compiler. For one particular case, the solution was to revert to
+   llvm version 9.0.1 using `Homebrew <https://brew.sh/>`_ and then add
+   environmental variables to your shell rc that point to the reverted
+   directory structure. ``numba`` also does not yet officially support
+   Python 3.9.
```

Comment:
Hopefully we can remove this after a @jhennawi PR that removes the numba dependence.